plutus-tx-plugin/test/Budget/9.6/lte0.pir.golden

```diff
@@ -1,39 +1,31 @@
-let
-  data Bool | Bool_match where
-    True : Bool
-    False : Bool
-in
 letrec
   data (List :: * -> *) a | List_match where
     Nil : List a
     Cons : a -> List a -> List a
 in
 letrec
-  !go : List integer -> Bool
+  !go : List integer -> bool
     = \(ds : List integer) ->
         List_match
           {integer}
           ds
-          {all dead. Bool}
-          (/\dead -> True)
+          {bool}
+          True
           (\(x : integer) (xs : List integer) ->
-             /\dead ->
-               Bool_match
-                 (case Bool (lessThanEqualsInteger x 0) [False, True])
-                 {all dead. Bool}
-                 (/\dead -> go xs)
-                 (/\dead -> False)
-                 {all dead. dead})
-          {all dead. dead}
+             case
+               (all dead. bool)
+               (lessThanEqualsInteger x 0)
+               [(/\dead -> False), (/\dead -> go xs)]
+               {all dead. dead})
 in
 letrec
   !go : integer -> List integer
     = \(n : integer) ->
-        Bool_match
-          (case Bool (lessThanEqualsInteger n 0) [False, True])
-          {all dead. List integer}
-          (/\dead -> Nil {integer})
-          (/\dead -> Cons {integer} 0 (go (subtractInteger n 1)))
+        case
+          (all dead. List integer)
+          (lessThanEqualsInteger n 0)
+          [ (/\dead -> Cons {integer} 0 (go (subtractInteger n 1)))
```

Comment:
The `/\dead ->` is no longer needed since `case` is already non-strict on the branches. It's Ok to leave this for a separate PR.

Reply:
I tried, this but it doesn't improve much. And actually hurt in some cases. It's probably messing with optimization. I'll skip this for now

Reply:
It doesn't really matter, because it's optimized away in UPLC anyway.

Reply:
`UntypedPlutusCore.Transform.ForceCaseDelay`

Reply:
Ah right, the latest optimization!